index/scorch/segment/zap/docvalues.go

```diff
@@ -178,3 +179,24 @@ func (s *Segment) VisitDocumentFieldTerms(localDocNum uint64, fields []string,
 	}
 	return nil
 }
+
+// VisitableDocValueFields returns the list of fields with
+// persisted doc value terms ready to be visitable using the
+// VisitDocumentFieldTerms method.
+func (s *Segment) VisitableDocValueFields() ([]string, error) {
+	if len(s.fieldsInv) == 0 {
```

Comment:
seems like it isn't necessary to treat this as an error, if there are no fields, none of them are visitable for doc values, so can't we just return nil